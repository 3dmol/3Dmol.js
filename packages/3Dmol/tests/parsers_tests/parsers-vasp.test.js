/**
*@jest-environment jsdom
*/

global.$ = require("jquery");
global.URL.createObjectURL = function() {};
let $3Dmol = require("../../build/3Dmol.js");
const fs = require('fs');

describe('Function VASP | Input: CONTCAR |', () => {
    const data = fs.readFileSync('../test_structs/CONTCAR', 'utf-8');
    let atoms = $3Dmol.Parsers.VASP(data);
    
    test("Atom is empty if input has less than 3 lines", () => {
        let str = "data";
        let result = $3Dmol.Parsers.VASP(str);
        expect(result).toEqual([[]]);
    });

    test("Atom is empty if second line of input is not a number", () => {
        let str = "line\nline\nline";
        let result = $3Dmol.Parsers.VASP(str);
        expect(result).toEqual([[]]);
    });

    test("Atoms is empty if second line of input is negative", () => {
        let str = "line1\n-1\nline3";
        let result = $3Dmol.Parsers.VASP(str);
        expect(result).toEqual([[]]);
    });
    
    test("Atoms length is 1", () => {
        expect(atoms.length).toBe(1);
    });
    
    test("Atoms should match the snapshot", ()=>{
        expect(atoms).toMatchSnapshot();
    });

});