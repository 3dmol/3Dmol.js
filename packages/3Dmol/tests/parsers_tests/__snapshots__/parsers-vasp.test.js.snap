// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Function VASP | Input: CONTCAR | Atoms should match the snapshot 1`] = `
Array [
  Array [
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": -0.0022009348342383367,
      "y": -0.0022009348342383367,
      "z": -0.0022009348342383367,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 0.8892802327397015,
      "y": 0.8892802327397015,
      "z": 0.8892802327397015,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7805698329747879,
      "y": -0.003074977966912264,
      "z": 1.7805698329747879,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.672083392999176,
      "y": 0.8887239766994924,
      "z": 2.672083392999176,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": -0.003074977966912264,
      "y": 1.7805698329747879,
      "z": 1.7805698329747879,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 0.8887239766994924,
      "y": 2.672083392999176,
      "z": 2.672083392999176,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7820270857659832,
      "y": 1.7820270857659832,
      "z": 3.5623959285154565,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.6780195193306113,
      "y": 2.6780195193306113,
      "z": 4.4511237322990285,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7805698329747879,
      "y": 1.7805698329747879,
      "z": -0.003074977966912264,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.672083392999176,
      "y": 2.672083392999176,
      "z": 0.8887239766994924,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5623959285154565,
      "y": 1.7820270857659832,
      "z": 1.7820270857659832,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.4511237322990285,
      "y": 2.6780195193306113,
      "z": 2.6780195193306113,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7820270857659832,
      "y": 3.5623959285154565,
      "z": 1.7820270857659832,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.6780195193306113,
      "y": 4.4511237322990285,
      "z": 2.6780195193306113,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5678012595466027,
      "y": 3.5678012595466027,
      "z": 3.5678012595466027,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.464867641692702,
      "y": 4.464867641692702,
      "z": 4.464867641692702,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.563272939133247,
      "y": -0.006487816946039519,
      "z": 3.563272939133247,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.456555640302951,
      "y": 0.8910412761098082,
      "z": 4.456555640302951,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.347894213697376,
      "y": -0.004037878093574932,
      "z": 5.347894213697376,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.23959815277925,
      "y": 0.8888168897963186,
      "z": 6.23959815277925,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5853303710330957,
      "y": 1.7839394354756763,
      "z": 5.332357573914908,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.463792656297843,
      "y": 2.6790868505573098,
      "z": 6.237898289600474,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.348549989288473,
      "y": 1.7809253071489963,
      "z": 7.130811966596322,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.2384203942306655,
      "y": 2.6710394760854936,
      "z": 8.022699134595657,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.332357573914908,
      "y": 1.7839394354756763,
      "z": 3.5853303710330957,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.237898289600474,
      "y": 2.6790868505573098,
      "z": 4.463792656297843,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.130811966596322,
      "y": 1.7809253071489963,
      "z": 5.348549989288473,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.022699134595657,
      "y": 2.6710394760854936,
      "z": 6.2384203942306655,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.359552089230692,
      "y": 3.580980886654287,
      "z": 5.359552089230692,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.256813865280642,
      "y": 4.477912895626028,
      "z": 6.256813865280642,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.135269988009827,
      "y": 3.5670005485107965,
      "z": 7.135269988009827,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.025030891879162,
      "y": 4.45144988490894,
      "z": 8.025030891879162,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": -0.006487816946039519,
      "y": 3.563272939133247,
      "z": 3.563272939133247,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 0.8910412761098082,
      "y": 4.456555640302951,
      "z": 4.456555640302951,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7839394354756763,
      "y": 3.5853303710330957,
      "z": 5.332357573914908,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.6790868505573098,
      "y": 4.463792656297843,
      "z": 6.237898289600474,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": -0.004037878093574932,
      "y": 5.347894213697376,
      "z": 5.347894213697376,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 0.8888168897963186,
      "y": 6.23959815277925,
      "z": 6.23959815277925,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7809253071489963,
      "y": 5.348549989288473,
      "z": 7.130811966596322,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.6710394760854936,
      "y": 6.2384203942306655,
      "z": 8.022699134595657,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7839394354756763,
      "y": 5.332357573914908,
      "z": 3.5853303710330957,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.6790868505573098,
      "y": 6.237898289600474,
      "z": 4.463792656297843,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.580980886654287,
      "y": 5.359552089230692,
      "z": 5.359552089230692,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.477912895626028,
      "y": 6.256813865280642,
      "z": 6.256813865280642,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 1.7809253071489963,
      "y": 7.130811966596322,
      "z": 5.348549989288473,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 2.6710394760854936,
      "y": 8.022699134595657,
      "z": 6.2384203942306655,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5670005485107965,
      "y": 7.135269988009827,
      "z": 7.135269988009827,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.45144988490894,
      "y": 8.025030891879162,
      "z": 8.025030891879162,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.56530508257522,
      "y": 3.56530508257522,
      "z": 7.128149910695151,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.453467927605795,
      "y": 4.453467927605795,
      "z": 8.024518487844034,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.339993307621755,
      "y": 3.5550119518274763,
      "z": 8.914980173331536,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.234785488241016,
      "y": 4.433416558417056,
      "z": 9.82110356256825,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5550119518274763,
      "y": 5.339993307621755,
      "z": 8.914980173331536,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.433416558417056,
      "y": 6.234785488241016,
      "z": 9.82110356256825,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.3404069939830965,
      "y": 5.3404069939830965,
      "z": 10.701738146601969,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.236648108801075,
      "y": 6.236648108801075,
      "z": 11.590992599678076,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.372318178641977,
      "y": 5.372318178641977,
      "z": 7.1589171153073154,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.201050911464427,
      "y": 6.201050911464427,
      "z": 8.14644853616317,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.1183207893914355,
      "y": 5.31730522699751,
      "z": 8.954494816455417,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.034439778089338,
      "y": 6.198696829523788,
      "z": 9.835440669994291,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.31730522699751,
      "y": 7.1183207893914355,
      "z": 8.954494816455417,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.198696829523788,
      "y": 8.034439778089338,
      "z": 9.835440669994291,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.128022660149498,
      "y": 7.128022660149498,
      "z": 10.701923547564055,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.017672897504191,
      "y": 8.017672897504191,
      "z": 11.586398397856065,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.563272939133247,
      "y": 3.563272939133247,
      "z": -0.006487816946039519,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.456555640302951,
      "y": 4.456555640302951,
      "z": 0.8910412761098082,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.332357573914908,
      "y": 3.5853303710330957,
      "z": 1.7839394354756763,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.237898289600474,
      "y": 4.463792656297843,
      "z": 2.6790868505573098,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5853303710330957,
      "y": 5.332357573914908,
      "z": 1.7839394354756763,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.463792656297843,
      "y": 6.237898289600474,
      "z": 2.6790868505573098,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.359552089230692,
      "y": 5.359552089230692,
      "z": 3.580980886654287,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.256813865280642,
      "y": 6.256813865280642,
      "z": 4.477912895626028,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.347894213697376,
      "y": 5.347894213697376,
      "z": -0.004037878093574932,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.23959815277925,
      "y": 6.23959815277925,
      "z": 0.8888168897963186,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.130811966596322,
      "y": 5.348549989288473,
      "z": 1.7809253071489963,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.022699134595657,
      "y": 6.2384203942306655,
      "z": 2.6710394760854936,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.348549989288473,
      "y": 7.130811966596322,
      "z": 1.7809253071489963,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.2384203942306655,
      "y": 8.022699134595657,
      "z": 2.6710394760854936,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.135269988009827,
      "y": 7.135269988009827,
      "z": 3.5670005485107965,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.025030891879162,
      "y": 8.025030891879162,
      "z": 4.45144988490894,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.128149910695151,
      "y": 3.56530508257522,
      "z": 3.56530508257522,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.024518487844034,
      "y": 4.453467927605795,
      "z": 4.453467927605795,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.914980173331536,
      "y": 3.5550119518274763,
      "z": 5.339993307621755,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.82110356256825,
      "y": 4.433416558417056,
      "z": 6.234785488241016,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.1589171153073154,
      "y": 5.372318178641977,
      "z": 5.372318178641977,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.14644853616317,
      "y": 6.201050911464427,
      "z": 6.201050911464427,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.954494816455417,
      "y": 5.31730522699751,
      "z": 7.1183207893914355,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.835440669994291,
      "y": 6.198696829523788,
      "z": 8.034439778089338,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.914980173331536,
      "y": 5.339993307621755,
      "z": 3.5550119518274763,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.82110356256825,
      "y": 6.234785488241016,
      "z": 4.433416558417056,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 10.701738146601969,
      "y": 5.3404069939830965,
      "z": 5.3404069939830965,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 11.590992599678076,
      "y": 6.236648108801075,
      "z": 6.236648108801075,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.954494816455417,
      "y": 7.1183207893914355,
      "z": 5.31730522699751,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.835440669994291,
      "y": 8.034439778089338,
      "z": 6.198696829523788,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 10.701923547564055,
      "y": 7.128022660149498,
      "z": 7.128022660149498,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 11.586398397856065,
      "y": 8.017672897504191,
      "z": 8.017672897504191,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.56530508257522,
      "y": 7.128149910695151,
      "z": 3.56530508257522,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.453467927605795,
      "y": 8.024518487844034,
      "z": 4.453467927605795,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.372318178641977,
      "y": 7.1589171153073154,
      "z": 5.372318178641977,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.201050911464427,
      "y": 8.14644853616317,
      "z": 6.201050911464427,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 3.5550119518274763,
      "y": 8.914980173331536,
      "z": 5.339993307621755,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 4.433416558417056,
      "y": 9.82110356256825,
      "z": 6.234785488241016,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.31730522699751,
      "y": 8.954494816455417,
      "z": 7.1183207893914355,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.198696829523788,
      "y": 9.835440669994291,
      "z": 8.034439778089338,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.339993307621755,
      "y": 8.914980173331536,
      "z": 3.5550119518274763,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.234785488241016,
      "y": 9.82110356256825,
      "z": 4.433416558417056,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.1183207893914355,
      "y": 8.954494816455417,
      "z": 5.31730522699751,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.034439778089338,
      "y": 9.835440669994291,
      "z": 6.198696829523788,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 5.3404069939830965,
      "y": 10.701738146601969,
      "z": 5.3404069939830965,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 6.236648108801075,
      "y": 11.590992599678076,
      "z": 6.236648108801075,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.128022660149498,
      "y": 10.701923547564055,
      "z": 7.128022660149498,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.017672897504191,
      "y": 11.586398397856065,
      "z": 8.017672897504191,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.951096578408341,
      "y": 7.007443253586856,
      "z": 8.951096578408341,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.780555606742825,
      "y": 7.994352560663962,
      "z": 9.780555606742825,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.007443253586856,
      "y": 8.951096578408341,
      "z": 8.951096578408341,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 7.994352560663962,
      "y": 9.780555606742825,
      "z": 9.780555606742825,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.896254888776445,
      "y": 8.896254888776445,
      "z": 10.675529424360404,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.793169250638263,
      "y": 9.793169250638263,
      "z": 11.572192261751866,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.951096578408341,
      "y": 8.951096578408341,
      "z": 7.007443253586856,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.780555606742825,
      "y": 9.780555606742825,
      "z": 7.994352560663962,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 10.675529424360404,
      "y": 8.896254888776445,
      "z": 8.896254888776445,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 11.572192261751866,
      "y": 9.793169250638263,
      "z": 9.793169250638263,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 8.896254888776445,
      "y": 10.675529424360404,
      "z": 8.896254888776445,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 9.793169250638263,
      "y": 11.572192261751866,
      "z": 9.793169250638263,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 10.68711273217525,
      "y": 10.68711273217525,
      "z": 10.68711273217525,
    },
    Object {
      "bonds": Array [],
      "elem": "C",
      "x": 11.584301155780338,
      "y": 11.584301155780338,
      "z": 11.584301155780338,
    },
    Object {
      "bonds": Array [],
      "elem": "Ge",
      "x": 7.574403222735554,
      "y": 7.574403222735554,
      "z": 7.574403222735554,
    },
  ],
]
`;
